### ISSUE TITLE
Task assembly may not be reported correctly when multiple tasks of the same name are used

### ISSUE THREAD

**Issue by ladipro** [KirillOsenkov/MSBuildStructuredLog#706]
The prime example is the `GenerateResource` task which lives in two assemblies: `Microsoft.Build.Tasks.Core` and `Microsoft.Build.Tasks.v3.5`. The current approach of keeping track of task assemblies by the bare task name only may lead to an incorrect assembly being reported by the viewer.

The relevant data structure:
https://github.com/KirillOsenkov/MSBuildStructuredLog/blob/7ed4f85d227cc1821990b695bb61f37c25efac85/src/StructuredLogger/Construction/Construction.cs#L23-L24

MSBuild very likely does not provide enough details to fix this without first enhancing the log output in https://github.com/dotnet/msbuild/issues/9290.

**Comment by KirillOsenkov** [comment]
Yes, I suspect once MSBuild logs a better message this issue will be fixed for free. There are also now messages when an assembly is loaded during task execution, perhaps we can correlate the full path of the assembly with the assembly name using those events.
